TCMSApp/src/client/app/apis/trello/create-project.service.js

```diff
@@ -0,0 +1,55 @@
+(function () {
+    'use strict';
+
+    angular
+        .module('app.trello')
+        .factory('createProjectFactory', createProjectFactory);
+
+    createProjectFactory.$inject = ['Trello', 'logger', 'addProjectFactory', 'fillSidebarFactory'];
+    /* @ngInject */
+    function createProjectFactory(Trello, logger, addProjectFactory, fillSidebarFactory) {
+
+        return {
+            createProjAndOrg: createProjAndOrg
+        };
+
+        function createProjAndOrg (Trello, projectName, projectDescription, vm, modalWindow) {
+
+            Trello.get('members/me/organizations').then(
+                function (res) {
+                    var same = false,
+                        l = res.length,
+                        i;
+                    for (i = 0; i < l; i++) {
+                        if (res[i].displayName === projectName) {
+                            same = true;
+                            break;
+                        }
+                    }
+                    if (same) logger.error('Project with the same name already exists.', '', 'Error');
```

Comment:
lines 20-28 (and ` logger.error('Project with the same name already exists.', '', 'Error');`) could be the separate function called `noSuchProject` returning true if no such project and logging error if there is and called as:

```js
if (noSuchProject()) {
   Trello.post('/organizations', {
      ...
}
```